llvm/lib/Target/SPIRV/SPIRVGlobalRegistry.cpp

```diff
@@ -726,8 +736,18 @@ static std::string buildSpirvTypeName(const SPIRVType *Type,
     }
     return TypeName;
   }
+  case SPIRV::OpTypeArray: {
+    Register ElementTypeReg = Type->getOperand(1).getReg();
+    auto *ElementType = MIRBuilder.getMRI()->getUniqueVRegDef(ElementTypeReg);
+    uint32_t ArraySize = 32; // Type->getOperand(2).getCImm()->getZExtValue();
```

Comment:
Is this expected?

Reply:
This is wrong. I've fixed it up, and added a test to make sure that two resoures with different array sizes are considered different variables.